### ISSUE TITLE
[Enhance]

### ISSUE THREAD

**Issue by mikechengwei** [pingcap/tidb-operator#4848]
## Feature Request

**Is your feature request related to a problem? Please describe:**

**Describe the feature you'd like:**

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

**Comment by csuzhangxc** [comment]
sounds good. Do you have more ideas about `tidb operator own update logic`?

**Comment by mikechengwei** [comment]
@csuzhangxc 
Own update logic:
1.A group of pd nodes upgrade only need to  evict leader once.
2.Large clusters can update nodes concurrently without affecting the health of the cluster
3.Check that upgrading nodes won't let the cluster into a worse situation
4.No need to maintain the `advancests`, add advancests feature into union crd.

Others:
1.The operation of pod control is more detailed, and it can complete such as in-place upgrade feature

**Comment by csuzhangxc** [comment]
👍 LGTM.

But more explanation about `4.No need to maintain the advancests, add advancests feature into union crd.` I think this feature can't replace delete-slots in asts.

**Comment by mikechengwei** [comment]
> 👍 LGTM.
> 
> But more explanation about `4.No need to maintain the advancests, add advancests feature into union crd.` I think this feature can't replace delete-slots in asts.

yes, you are right. we must use `podset` crd instead of sts crd like kafka operator.
